core/src/main/scala/org/scalatra/servlet/RichServletContext.scala

```diff
@@ -222,5 +222,17 @@ case class RichServletContext(sc: ServletContext) extends AttributesMap {
 
   def contextPath: String = sc.getContextPath
 
+  val generatedUrlsKey = "org.scalatra.scalate.generatedUrls"
+
+  def addGeneratedUrls(servletClassName: String, routes: Map[String, Route]) = {
+    sc.setAttribute(generatedUrlsKey, generatedUrls + (servletClassName -> routes))
```

Comment:
It would be nice if it warns name collision as you mentioned in the gitter room.

Reply:
Ok, I can add in a Logger instance and a warning.